README.md
# Diablo2Kits

## English

D2 Barbarian Kit

D2 Paladin Kit

## 한국어

디아블로2의 직업 중 바바리안, 팔라딘을 키트로 구현하였습니다.  

[D2 바바리안 키트](https://github.com/hybridgame/Diablo2Kits/blob/master/d2%23bar/README%20Korean.md)

[D2 팔라딘 키트](https://github.com/hybridgame/Diablo2Kits/blob/master/d2%23pal/README%20Korean.md)



doc/ChangeLog.md
# Change Log

## 2022-11-05

### D2 Paladin

- 성능이 낮은 오라들 상향
  - 마이트 공격력 2~5 -> 2~8
  - 컨센트레이션 공격력 4~7 -> 4~10
  - 블레스드 에임 THAC0 4~10 -> 3~12
  - 레지스트 파이어, 라이트닝, 콜드 저항 20%~50% -> 20%~80%
  - 샐베이션 원소 저항 20%~50% -> 15%~60%
  - 비거 이동 속도 1~4 -> 2~8
  - 디파이언스 방어력 1~4 -> 2~8
  - 프레이어 체력회복 1/5초~1/2초 -> 1/4초~1/1초

## 2022-10-26

- Readme html, pdf 포맷 문서 삭제

### D2 Paladin

- 컴뱃 스킬 교체시 새크리파이스 효과를 제거하는 효과를 삭제

## 2019-10-13

### D2 Barbarian

- 스크립트와 다이얼로그 최적화

### D2 Paladin

- 스크립트와 다이얼로그 최적화

## 2019-10-05

### D2 Barbarian

- D2 바바리안 코간 추가

### D2 Paladin

- D2 팔라딘 켈돈 추가

## 2019-07-10

### D2 Paladin

- 새크리파이스 스킬 제거

## 2019-07-05

### D2 Barbarian

- 배틀 오더와 프렌지 텍스트 수정

## 2019-07-04

- 키드 설명 수정
- X 키를 누르면 스킬을 선택할 수 있게 변경

### D2 Paladin

- 캐릭터에 오라 애니메이션 추가
- 파나티시즘 데미지 추가를 8에서 5로 변경
- 오라의 효과를 D2 팔라딘 본인에게만 적용되도록 변경

## 2019-07-03

- 키트 설명에 스킬 트리 추가

### D2 Paladin

- 홀리 볼트가 파티원을 치료하는 효과가 적용되지 않는 것을 수정

d2#bar/baf/d2#bar.d
BEGIN D2#BAR

IF ~True()~ selectSkill
    SAY @17
    + ~True()~ + @1101 + howl
    + ~!Global("D2#1101","LOCALS",0) LevelGT(Myself,3)~ + @1103 + shout
    + ~!Global("D2#1101","LOCALS",0) LevelGT(Myself,3)~ + @1104 + taunt
    + ~!Global("D2#1104","LOCALS",0) LevelGT(Myself,11)~ + @1105 + battleCry
    + ~!Global("D2#1103","LOCALS",0) LevelGT(Myself,15)~ + @1107 + battleOrder
    + ~Global("D2#1111","LOCALS",0)
        Global("D2#1114","LOCALS",0)
        Global("D2#1115","LOCALS",0)
        LevelGT(Myself,3)~ + @1111 + twsm
    + ~Global("D2#1111","LOCALS",0)
        Global("D2#1114","LOCALS",0)
        Global("D2#1115","LOCALS",0)
        LevelGT(Myself,3)~ + @1114 + thwsm
    + ~Global("D2#1111","LOCALS",0)
        Global("D2#1114","LOCALS",0)
        GlobalLT("D2#1115","LOCALS",5)
        LevelGT(Myself,3)~ + @1115 + throwingMastery
    + ~OR(3)
            !Global("D2#1111","LOCALS",0)
            !Global("D2#1114","LOCALS",0)
            !Global("D2#1115","LOCALS",0)
        GlobalLT("D2#1118","LOCALS",5)
        LevelGT(Myself,11)~ + @1118 + ironSkin
    + ~OR(3)
            !Global("D2#1111","LOCALS",0)
            !Global("D2#1114","LOCALS",0)
            !Global("D2#1115","LOCALS",0)
            GlobalLT("D2#1119","LOCALS",5)
        LevelGT(Myself,15)~ + @1119 + increasedSpeed
    + ~Global("D2#1121","LOCALS",0)~ + @1121 + bash
    + ~Global("D2#1121","LOCALS",1) Global("D2#1122","LOCALS",0) LevelGT(Myself,3)~ + @1122 + doubleSwing
    + ~Global("D2#1122","LOCALS",1) Global("D2#1124","LOCALS",0) LevelGT(Myself,7)~ + @1124 + doubleThrow
    + ~Global("D2#1121","LOCALS",1) Global("D2#1125","LOCALS",0) LevelGT(Myself,7)~ + @1125 + stun
    + ~Global("D2#1125","LOCALS",1) Global("D2#1127","LOCALS",0) LevelGT(Myself,11)~ + @1127 + concentration
    + ~Global("D2#1124","LOCALS",1) Global("D2#1128","LOCALS",0) LevelGT(Myself,15)~ + @1128 + frenzy
    ++ @2 DO ~IncrementGlobal("D2#BARSkillPoint","LOCALS",1)~ EXIT
END

IF ~~ howl
    SAY @1131
    ++ @16 DO ~IncrementGlobal("D2#1101","LOCALS",1) ApplySpellRES("D2#1101L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ shout
    SAY @1133
    ++ @16 DO ~IncrementGlobal("D2#1103","LOCALS",1) ApplySpellRES("D2#1103L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ taunt
    SAY @1134
    ++ @16 DO ~IncrementGlobal("D2#1104","LOCALS",1) ApplySpellRES("D2#1104L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ battleCry
    SAY @1135
    ++ @16 DO ~IncrementGlobal("D2#1105","LOCALS",1) ApplySpellRES("D2#1105L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ battleOrder
    SAY @1137
    ++ @16 DO ~IncrementGlobal("D2#1107","LOCALS",1) ApplySpellRES("D2#1107L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ twsm
    SAY @1141
    ++ @16 DO ~IncrementGlobal("D2#1111","LOCALS",1) ApplySpellRES("D2#1111L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ thwsm
    SAY @1144
    ++ @16 DO ~IncrementGlobal("D2#1114","LOCALS",1) ApplySpellRES("D2#1114L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ throwingMastery
    SAY @1145
    ++ @16 DO ~IncrementGlobal("D2#1115","LOCALS",1) ApplySpellRES("D2#1115L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ ironSkin
    SAY @1148
    ++ @16 DO ~IncrementGlobal("D2#1118","LOCALS",1) ApplySpellRES("D2#1118L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ increasedSpeed
    SAY @1149
    ++ @16 DO ~IncrementGlobal("D2#1119","LOCALS",1) ApplySpellRES("D2#1119L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ bash
    SAY @1151
    ++ @16 DO ~IncrementGlobal("D2#1121","LOCALS",1) ApplySpellRES("D2#1121L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ doubleSwing
    SAY @1152
    ++ @16 DO ~IncrementGlobal("D2#1122","LOCALS",1) ApplySpellRES("D2#1122L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ doubleThrow
    SAY @1154
    ++ @16 DO ~IncrementGlobal("D2#1124","LOCALS",1) ApplySpellRES("D2#1124L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ stun
    SAY @1155
    ++ @16 DO ~IncrementGlobal("D2#1125","LOCALS",1) ApplySpellRES("D2#1125L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ concentration
    SAY @1157
    ++ @16 DO ~IncrementGlobal("D2#1127","LOCALS",1) ApplySpellRES("D2#1127L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ frenzy
    SAY @1158
    ++ @16 DO ~IncrementGlobal("D2#1128","LOCALS",1) ApplySpellRES("D2#1128L",Myself)~ EXIT
    ++ @2 + selectSkill
END

d2#pal/baf/d2#pal.d
BEGIN D2#PAL

IF ~True()~ selectSkill
    SAY @17
    + ~Global("D2#1201","LOCALS",0)~ + @1201 + prayer
    + ~Global("D2#1202","LOCALS",0)~ + @1202 + resistFire
    + ~Global("D2#1203","LOCALS",0) LevelGT(Myself,3)~ + @1203 + resistCold
    + ~Global("D2#1204","LOCALS",0) LevelGT(Myself,7)~ + @1204 + resistLightning
    + ~Global("D2#1205","LOCALS",0) LevelGT(Myself,3)~ + @1205 + defiance
    + ~Global("D2#1207","LOCALS",0) LevelGT(Myself,11)~ + @1207 + vigor
    + ~Global("D2#1211","LOCALS",0)~ + @1211 + might
    + ~Global("D2#1211","LOCALS",1) Global("D2#1212","LOCALS",0) LevelGT(Myself,3)~ + @1212 + holyFire
    + ~Global("D2#1211","LOCALS",1) Global("D2#1214","LOCALS",0) LevelGT(Myself,7)~ + @1214 + blessedAim
    + ~Global("D2#1214","LOCALS",1) Global("D2#1215","LOCALS",0) LevelGT(Myself,11)~ + @1215 + concentration
    + ~Global("D2#1212","LOCALS",1) Global("D2#1216","LOCALS",0) LevelGT(Myself,11)~ + @1216 + holyFreeze
    + ~Global("D2#1216","LOCALS",1) Global("D2#1217","LOCALS",0) LevelGT(Myself,14)~ + @1217 + holyShock
    + ~LevelGT(Myself,3)~ + @1223 + holyBolt
    + ~Global("D2#1224","LOCALS",0) LevelGT(Myself,7)~ + @1224 + zeal
    + ~Global("D2#1224","LOCALS",1) Global("D2#1226","LOCALS",0) LevelGT(Myself,11)~ + @1226 + vengeance
    + ~GlobalGT("D2#1223","LOCALS",0) LevelGT(Myself,11)~ + @1227 + blessedHammer
    + ~GlobalGT("D2#1227","LOCALS",0) LevelGT(Myself,14)~ + @1229 + holyShield
    ++ @2 DO ~IncrementGlobal("D2#PALSkillPoint","LOCALS",1)~ EXIT
END

IF ~~ prayer
    SAY @1231
    ++ @16 DO ~IncrementGlobal("D2#1201","LOCALS",1) ApplySpellRES("D2#1201L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ resistFire
    SAY @1232
    ++ @16 DO ~IncrementGlobal("D2#1202","LOCALS",1) ApplySpellRES("D2#1202L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ resistCold
    SAY @1233
    ++ @16 DO ~IncrementGlobal("D2#1203","LOCALS",1) ApplySpellRES("D2#1203L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ resistLightning
    SAY @1234
    ++ @16 DO ~IncrementGlobal("D2#1204","LOCALS",1) ApplySpellRES("D2#1204L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ defiance
    SAY @1235
    ++ @16 DO ~IncrementGlobal("D2#1205","LOCALS",1) ApplySpellRES("D2#1205L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ vigor
    SAY @1237
    ++ @16 DO ~IncrementGlobal("D2#1207","LOCALS",1) ApplySpellRES("D2#1207L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ might
    SAY @1241
    ++ @16 DO ~IncrementGlobal("D2#1211","LOCALS",1) ApplySpellRES("D2#1211L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ holyFire
    SAY @1242
    ++ @16 DO ~IncrementGlobal("D2#1212","LOCALS",1) ApplySpellRES("D2#1212L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ blessedAim
    SAY @1244
    ++ @16 DO ~IncrementGlobal("D2#1214","LOCALS",1) ApplySpellRES("D2#1214L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ concentration
    SAY @1245
    ++ @16 DO ~IncrementGlobal("D2#1215","LOCALS",1) ApplySpellRES("D2#1215L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ holyFreeze
    SAY @1246
    ++ @16 DO ~IncrementGlobal("D2#1216","LOCALS",1) ApplySpellRES("D2#1216L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ holyShock
    SAY @1247
    ++ @16 DO ~IncrementGlobal("D2#1217","LOCALS",1) ApplySpellRES("D2#1217L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ holyBolt
    SAY @1253
    ++ @16 DO ~IncrementGlobal("D2#1223","LOCALS",1) ApplySpellRES("D2#1223L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ zeal
    SAY @1254
    ++ @16 DO ~IncrementGlobal("D2#1224","LOCALS",1) ApplySpellRES("D2#1224L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ vengeance
    SAY @1256
    ++ @16 DO ~IncrementGlobal("D2#1226","LOCALS",1) ApplySpellRES("D2#1226L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ blessedHammer
    SAY @1257
    ++ @16 DO ~IncrementGlobal("D2#1227","LOCALS",1) ApplySpellRES("D2#1227L",Myself)~ EXIT
    ++ @2 + selectSkill
END

IF ~~ holyShield
    SAY @1259
    ++ @16 DO ~IncrementGlobal("D2#1229","LOCALS",1) ApplySpellRES("D2#1229L",Myself)~ EXIT
    ++ @2 + selectSkill
END

d2#pal/baf/d2#palsp.d
BEGIN D2#PALSP

IF ~NumTimesTalkedTo(0)~ THEN BEGIN selectSkill
    SAY @104

    IF ~~ THEN REPLY @2
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#PALSkillPoint","GLOBAL",1))~ EXIT

    // Defensive Aura

    // Prayer
    IF ~Global("D2#1201","GLOBAL",0)~ THEN REPLY @1201
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1201","GLOBAL",1))
            ApplySpellRES("D2#1201L",LastSeenBy(Myself))~ EXIT

    // Resist Fire
    IF ~Global("D2#1202","GLOBAL",0)~ THEN REPLY @1202
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1202","GLOBAL",1))
            ApplySpellRES("D2#1202L",LastSeenBy(Myself))~ EXIT

    // Resist Cold
    IF ~Global("D2#1203","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1203
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1203","GLOBAL",1))
            ApplySpellRES("D2#1203L",LastSeenBy(Myself))~ EXIT

    // Resist Lightning
    IF ~Global("D2#1204","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),7)~ THEN REPLY @1204
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1204","GLOBAL",1))
            ApplySpellRES("D2#1204L",LastSeenBy(Myself))~ EXIT

    // Defiance
    IF ~Global("D2#1205","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1205
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1205","GLOBAL",1))
            ApplySpellRES("D2#1205L",LastSeenBy(Myself))~ EXIT

    // Vigor
    IF ~Global("D2#1207","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1207
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1207","GLOBAL",1))
            ApplySpellRES("D2#1207L",LastSeenBy(Myself))~ EXIT

    // Offensive Aura

    // Might
    IF ~Global("D2#1211","GLOBAL",0)~ THEN REPLY @1211
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1211","GLOBAL",1))
            ApplySpellRES("D2#1211L",LastSeenBy(Myself))~ EXIT

    // Holy Fire
    IF ~Global("D2#1211","GLOBAL",1)
        Global("D2#1212","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1212
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1212","GLOBAL",1))
            ApplySpellRES("D2#1212L",LastSeenBy(Myself))~ EXIT

    // Blessed Aim
    IF ~Global("D2#1211","GLOBAL",1)
        Global("D2#1214","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),7)~ THEN REPLY @1214
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1214","GLOBAL",1))
            ApplySpellRES("D2#1214L",LastSeenBy(Myself))~ EXIT

    // Concentration
    IF ~Global("D2#1214","GLOBAL",1)
        Global("D2#1215","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1215
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1215","GLOBAL",1))
            ApplySpellRES("D2#1215L",LastSeenBy(Myself))~ EXIT

    // Holy Freeze
    IF ~Global("D2#1212","GLOBAL",1)
        Global("D2#1216","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1216
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1216","GLOBAL",1))
            ApplySpellRES("D2#1216L",LastSeenBy(Myself))~ EXIT

    // Holy Shock
    IF ~Global("D2#1216","GLOBAL",1)
        Global("D2#1217","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),14)~ THEN REPLY @1217
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1217","GLOBAL",1))
            ApplySpellRES("D2#1217L",LastSeenBy(Myself))~ EXIT

    // Combat Skill

    // Holy Bolt
    IF ~LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1223
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1223","GLOBAL",1))
            ApplySpellRES("D2#1223L",LastSeenBy(Myself))~ EXIT

    // Zeal
    IF ~Global("D2#1224","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),7)~ THEN REPLY @1224
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1224","GLOBAL",1))
            ApplySpellRES("D2#1224L",LastSeenBy(Myself))~ EXIT

    // Vengeance
    IF ~Global("D2#1224","GLOBAL",1)
        Global("D2#1226","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1226
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1226","GLOBAL",1))
            ApplySpellRES("D2#1226L",LastSeenBy(Myself))~ EXIT

    // Blessed Hammer
    IF ~GlobalGT("D2#1223","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1227
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1227","GLOBAL",1))
            ApplySpellRES("D2#1227L",LastSeenBy(Myself))~ EXIT

    // Holy Shield
    IF ~GlobalGT("D2#1227","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),14)~ THEN REPLY @1229
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1229","GLOBAL",1))
            ApplySpellRES("D2#1229L",LastSeenBy(Myself))~ EXIT

END

d2#bar/baf/d2#barsp.d
BEGIN D2#BARSP

IF ~True()~ THEN BEGIN selectSkill
    SAY @104

    IF ~~ THEN REPLY @2
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#BARSkillPoint","GLOBAL",1))~ EXIT

    // Howl
    IF ~~ THEN REPLY @1101
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1101","GLOBAL",1))
            ApplySpellRES("D2#1101L",LastSeenBy(Myself))~ EXIT

    // Shout
    IF ~!Global("D2#1101","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1103
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1103","GLOBAL",1))
            ApplySpellRES("D2#1103L",LastSeenBy(Myself))~ EXIT

    // Taunt
    IF ~!Global("D2#1101","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1104
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1104","GLOBAL",1))
            ApplySpellRES("D2#1104L",LastSeenBy(Myself))~ EXIT

    // Battle cry
    IF ~!Global("D2#1104","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1105
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1105","GLOBAL",1))
            ApplySpellRES("D2#1105L",LastSeenBy(Myself))~ EXIT

    // Battle Order
    IF ~!Global("D2#1103","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),15)~ THEN REPLY @1107
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1107","GLOBAL",1))
            ApplySpellRES("D2#1107L",LastSeenBy(Myself))~ EXIT
    
    // Two Weapon Style Mastery
    IF ~Global("D2#1111","GLOBAL",0)
        Global("D2#1114","GLOBAL",0)
        Global("D2#1115","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1111
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1111","GLOBAL",1))
            ApplySpellRES("D2#1111L",LastSeenBy(Myself))~ EXIT

    // Two Handed Weapon Style Mastery
    IF ~Global("D2#1111","GLOBAL",0)
        Global("D2#1114","GLOBAL",0)
        Global("D2#1115","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1114
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1114","GLOBAL",1))
            ApplySpellRES("D2#1114L",LastSeenBy(Myself))~ EXIT

    // Throwing Mastery
    IF ~Global("D2#1111","GLOBAL",0)
        Global("D2#1114","GLOBAL",0)
        GlobalLT("D2#1115","GLOBAL",5)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1115
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1115","GLOBAL",1))
            ApplySpellRES("D2#1115L",LastSeenBy(Myself))~ EXIT

    // Iron Skin
    IF ~OR(3)
            !Global("D2#1111","GLOBAL",0)
            !Global("D2#1114","GLOBAL",0)
            !Global("D2#1115","GLOBAL",0)
        GlobalLT("D2#1118","GLOBAL",5)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1118
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1118","GLOBAL",1))
            ApplySpellRES("D2#1118L",LastSeenBy(Myself))~ EXIT

    // Increased Speed
    IF ~OR(3)
            !Global("D2#1111","GLOBAL",0)
            !Global("D2#1114","GLOBAL",0)
            !Global("D2#1115","GLOBAL",0)
            GlobalLT("D2#1119","GLOBAL",5)
        LevelGT(LastSeenBy(Myself),15)~ THEN REPLY @1119
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1119","GLOBAL",1))
            ApplySpellRES("D2#1119L",LastSeenBy(Myself))~ EXIT

    // Bash
    IF ~Global("D2#1121","GLOBAL",0)~ THEN REPLY @1121
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1121","GLOBAL",1))
            ApplySpellRES("D2#1121L",LastSeenBy(Myself))~ EXIT

    // Double Swing
    IF ~Global("D2#1121","GLOBAL",1)
        Global("D2#1122","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),3)~ THEN REPLY @1122
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1122","GLOBAL",1))
            ApplySpellRES("D2#1122L",LastSeenBy(Myself))~ EXIT

    // Double Throw
    IF ~Global("D2#1122","GLOBAL",1)
        Global("D2#1124","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),7)~ THEN REPLY @1124
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1124","GLOBAL",1))
            ApplySpellRES("D2#1124L",LastSeenBy(Myself))~ EXIT

    // Stun
    IF ~Global("D2#1121","GLOBAL",1)
        Global("D2#1125","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),7)~ THEN REPLY @1125
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1125","GLOBAL",1))
            ApplySpellRES("D2#1125L",LastSeenBy(Myself))~ EXIT

    // Concentration
    IF ~Global("D2#1125","GLOBAL",1)
        Global("D2#1127","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),11)~ THEN REPLY @1127
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1127","GLOBAL",1))
            ApplySpellRES("D2#1127L",LastSeenBy(Myself))~ EXIT

    // Frenzy
    IF ~Global("D2#1124","GLOBAL",1)
        Global("D2#1128","GLOBAL",0)
        LevelGT(LastSeenBy(Myself),15)~ THEN REPLY @1128
        DO ~ActionOverride(LastSeenBy(Myself),IncrementGlobal("D2#1128","GLOBAL",1))
            ApplySpellRES("D2#1128L",LastSeenBy(Myself))~ EXIT
END